### PR TITLE
Adding links to chainspec file and gas limit details

### PR DESCRIPTION
**Pull request by Buddhika-Pathirathna**
### Related links

Related card - [634](https://github.com/casper-network/docs/issues/634)

### Changes
- Added the link to the chainspec file in the Glossary definition
- link the glossary definition to the 'chainspec' word occurrences
- Added the link to the gas limit in required occurances

### CODE REVIEW

File: source/docs/casper/design/execution-semantics.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ The Casper Network is a decentralized computation platform. In this chapter we d
...
+Computation is all done in a [WebAssembly (wasm)](https://webassembly.org/) interpreter, allowing any programming language which compiles to wasm to become a smart contract language for the Casper blockchain. Similar to Ethereum, we use [`Gas`](/economics/gas-concepts/) to measure computational work in a way which is consistent from node to node in the Casper Network. Each wasm instruction is [assigned](https://github.com/casper-network/casper-node/blob/cb1d20ad1ea6e245cd8237f9406885a1e785c669/execution_engine/src/shared/wasm_config.rs#L15) a `Gas` value, and the amount of gas spent is tracked by the runtime with each instruction executed by the interpreter. All executions are finite because each has a finite [_gas limit_](https://github.com/casper-network/casper-node/blob/b94c4f79ac4ca00e996c418dcc3a98607779a193/resources/production/chainspec.toml#L96-L97) that specifies the maximum amount of gas that can be spent before the computation is terminated by the runtime. How this limit is determined is discussed in more detail below.
```
Comment (ipopescu) [inline source/docs/casper/design/execution-semantics.md:11]: Here we are pointing to the block gas limit, which may not be the same. Do we need to reword this to say: "All executions are finite because each block has a finite gas limit", @darthsiroftardis? 
```suggestion
Computation is all done in a [WebAssembly (wasm)](https://webassembly.org/) interpreter, allowing any programming language which compiles to wasm to become a smart contract language for the Casper blockchain. Similar to Ethereum, we use [`Gas`](/economics/gas-concepts/) to measure computational work in a way which is consistent from node to node in the Casper Network. Each wasm instruction is [assigned](https://github.com/casper-network/casper-node/blob/cb1d20ad1ea6e245cd8237f9406885a1e785c669/execution_engine/src/shared/wasm_config.rs#L15) a `Gas` value, and the amount of gas spent is tracked by the runtime with each instruction executed by the interpreter. All executions are finite because each block has a finite [_gas limit_](https://github.com/casper-network/casper-node/blob/b94c4f79ac4ca00e996c418dcc3a98607779a193/resources/production/chainspec.toml#L96-L97) that specifies the maximum amount of gas that can be spent per block before the computation is terminated by the runtime. How this limit is determined is discussed in more detail below.
```

File: source/docs/casper/economics/gas-concepts.md [hunk 2]
```diff
@@ -48,4 +48,4 @@ Currently, we are hard at work to create tools to help you estimate gas costs. M
...
+When a smart contract hits its [gas limit](https://github.com/casper-network/casper-node/blob/b94c4f79ac4ca00e996c418dcc3a98607779a193/resources/production/chainspec.toml#L96-L97) (the payment amount), execution stops. If your limit is 2.5 CSPR, execution stops and that is the computation cost even if the smart contract did not run to completion. So, the error message tries to communicate to you that execution stopped at 2.5 CSPR. The computation resulted in an error because there were not enough funds to run to completion. It would have cost more than 2.5 CSPR to complete execution, but since you only supplied a payment of 2.5 CSPR worth of computation, the network stopped execution there and charged you that much, even though it was a failed transaction. The execution engine does not actually know how much it would have cost if allowed to run to completion, because it did not allow the contract to finish since the contract would have run over its gas limit.
```
Comment (ipopescu) [inline source/docs/casper/economics/gas-concepts.md:51]: This is the block gas limit, not the contract's gas limit. Before changing this, we need a review from @darthsiroftardis.

File: source/docs/casper/economics/runtime.md [hunk 3]
```diff
@@ -19,13 +19,13 @@ Additionally, a minimal amount of CSPR must be present in the user account to en
...
 
 We expect to refine the current gas cost table to reflect time use more closely, with updates introduced in future upgrades. We also anticipate that, with the introduction of state pruning, storage costs will be calculated separately from computing time.
 
 ### Lanes {#lanes}
 
-The block gas limit is split into two lanes, one for native transfers and one for general deploys. The number of transfers, which cost a fixed amount of gas, is governed directly by the `block_max_transfer_count` chainspec parameter, set to 2500 when Mainnet launched.
+The [block gas limit](https://github.com/casper-network/casper-node/blob/b94c4f79ac4ca00e996c418dcc3a98607779a193/resources/production/chainspec.toml#L96-L97) is split into two lanes, one for native transfers and one for general deploys. The number of transfers, which cost a fixed amount of gas, is governed directly by the `block_max_transfer_count` chainspec parameter, set to 2500 when Mainnet launched.
```
Comment (ipopescu) [inline source/docs/casper/economics/runtime.md:28]: This is good here.

File: source/docs/casper/glossary/C.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ Correct-by-construction (CBC) protocols are consensus protocols meeting the foll
 
 ## Chainspec {#chainspec}
 
-A collection of configuration settings describing the state of the system at genesis and upgrades to basic system functionality (including system contracts and gas costs) occurring after [genesis](G.md#genesis).
+A collection of configuration settings describing the state of the system at genesis and upgrades to basic system functionality (including system contracts and gas costs) occurring after [genesis](G.md#genesis). Changes to the [chainspec](https://github.com/casper-network/casper-node/blob/release-1.4.8/resources/production/chainspec.toml) occurs depending on the system updates and releases.
```
Comment (ipopescu) [inline source/docs/casper/glossary/C.md:26]:
```suggestion
A collection of configuration settings describing the state of the system at genesis and upgrades to basic system functionality (including system contracts and gas costs) occurring after [genesis](G.md#genesis). Here is an example [chainspec](https://github.com/casper-network/casper-node/blob/release-1.4.8/resources/production/chainspec.toml), which will change with newer releases.
```

File: source/docs/casper/operators/create.md [hunk 5]
```diff
@@ -11,7 +11,7 @@ The Casper node software creates a genesis block from the following input files:
 
 ### chainspec.toml {#chainspectoml}
 
-A version of the chainspec is downloaded by the `pull_casper_node_version.sh` script installed with the casper-node-launcher debian package. This script pulls the chainspec.toml file from the appropriate path defined in the network config file used (`casper.conf` for MainNet and `casper-test.conf` for TestNet).
+A version of the [chainspec](https://github.com/casper-network/casper-node/blob/release-1.4.8/resources/production/chainspec.toml) is downloaded by the `pull_casper_node_version.sh` script installed with the casper-node-launcher debian package. This script pulls the chainspec.toml file from the appropriate path defined in the network config file used (`casper.conf` for MainNet and `casper-test.conf` for TestNet).
```
Comment (ipopescu) [inline source/docs/casper/operators/create.md:14]: Here you can point to the glossary so that the chainspec example is updated in a single location (the one from the glossary).

**Review by ipopescu — CHANGES_REQUESTED** [review]
Requesting a technical review from @darthsiroftardis. I think it's worth being exact about the wording and should be a quick check. Thanks!

File: source/docs/casper/design/execution-semantics.md (hunk 1 above)
Comment (darthsiroftardis) [reply source/docs/casper/design/execution-semantics.md:11]: Okay, so there is a block gas limit, but the payment amount specified within the deploy structure determines the the gas limit for execution. How much gas you're willing to pay is the limit, if this section is point to the block gas limit, then we should change it to not point there

File: source/docs/casper/economics/gas-concepts.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ Gas fees are consumed on the network irrespective of whether your transaction wa
 
 ## How is gas cost determined?
 
-The amount of gas required for a transaction is determined by how much code is executed on the blockchain. Currently, gas is priced at a fixed price of 1 mote (defined as 1x10^9 CSPR tokens) per 1 unit of gas. The gas charged for a transaction on the blockchain is paid to the validators who are operating the network.
+The amount of gas required for a transaction is determined by how much code is executed on the blockchain. Usually, the platform defines the required gas limit for a Deploy, inside a block through a configuration in the [chainspec](https://github.com/casper-network/casper-node/blob/release-1.4.8/resources/production/chainspec.toml) file. Currently, gas is priced at a fixed price of 1 mote (defined as 1x10^9 CSPR tokens) per 1 unit of gas. The gas charged for a transaction on the blockchain is paid to the validators who are operating the network.
```
Comment (darthsiroftardis) [inline source/docs/casper/economics/gas-concepts.md:13]: This sounds incorrect, the deploy sender sets the gas price which is shown in the deploy header structure, the only upper limit of gas is the block gas limit itself, and the information in the parentheses is just dead wrong, unless I am reading it wrong 1 CSPR is 10^9 motes, not the other way around

File: source/docs/casper/economics/gas-concepts.md (hunk 2 above)
Comment (darthsiroftardis) [reply source/docs/casper/economics/gas-concepts.md:51]: @ipopescu You are correct, these links are pointing to the block gas limit which is very different from what is being spoken about in this paragraph

**Review by darthsiroftardis — COMMENTED** [review]
Some corrections are needed before this PR can be approved

File: source/docs/casper/design/execution-semantics.md (hunk 1 above)
Comment (ipopescu) [reply source/docs/casper/design/execution-semantics.md:11]: Understood. So @Buddhika-Pathirathna there are 2 solutions here:
1) remove the link
2) remove the link and add a short additional explanation, which would clarify that the payment amount specified within the deploy sets the gas limit for deploy execution. 
I prefer option 2 since I was confused here :).

File: source/docs/casper/economics/gas-concepts.md (hunk 2 above)
Comment (ipopescu) [reply source/docs/casper/economics/gas-concepts.md:51]: Ok let's remove the link here and go back to what we had before @Buddhika-Pathirathna.

File: source/docs/casper/economics/gas-concepts.md (hunk 6 above)
Comment (ipopescu) [reply source/docs/casper/economics/gas-concepts.md:13]: There should be a negative sign there. Or we should say in brackets (1 CSPR is 10^9 motes). Let's remove the link and update the information about motes:
```suggestion
The amount of gas required for a transaction is determined by how much code is executed on the blockchain. Currently, gas is priced at a fixed price of 1 mote (1 CSPR is 10^9 motes) per 1 unit of gas. The gas charged for a transaction on the blockchain is paid to the network's validators.
```

**Comment by ipopescu** [comment]
@Buddhika-Pathirathna Let's use the link to the chainspec in the glossary. Please ping me when the PR is ready for another review pass. Thank you!

**Review by ipopescu — APPROVED** [review]
LGTM, @Buddhika-Pathirathna! Please merge when you can.
